src/include/btmem.h

```diff
@@ -128,6 +128,46 @@ __wt_page_header_byteswap(WT_PAGE_HEADER *dsk)
 #define WT_PAGE_HEADER_BYTE(btree, dsk) \
     ((void *)((uint8_t *)(dsk) + WT_PAGE_HEADER_BYTE_SIZE(btree)))
 
+/*
+ * WT_DELTA_HEADER --
+ *	Header for delta
+ */
+struct __wt_delta_header {
```

Comment:
I agree with Peter, having 4 magic bytes seems appropriate.

Reply:
We don't have this for the page header. Should we have this really?

Reply:
@quchenhao You're right - In WT, we have a page header WT_PAGE_HEADER followed by block header WT_BLOCK_HEADER.  We don't have magic number, but we do have a checksum in the block header which negates the need for a magic number.  I don't think we want to reuse either of those structs, they each contain things we don't need.  But from WT_BLOCK_HEADER, we at least need the checksum (4 bytes).

However, I think we need a way to distinguish the raw bytes of a full-page (page_header + block_header) from the raw bytes of a delta_header.  This could be to detect that something is really wrong - we expected a delta, but we got the bytes for a full-page, etc.  Or maybe we need something like salvage for raw bytes.  How to achieve this?  If we keep the page_header as is (for full page), the first eight bytes are: `uint64_t recno` - the first recno on the page for column store.  Even assuming we might someday support column store, I think we can safely use the upper-most byte to distinguish (that leaves us 2^56 recnos).  However the recno is stored as little endian, meaning the "top" byte is byte 7, that's the one that must be non-zero for a delta.

Maybe:
```
uint32_t mem_size; /* 00-03 */
uint8_t version;  /* 04 */
uint8_t type;   /* 05 */
uint8_t flags;  /* 06 */
uint8_t distinguished;  /* 07: cannot be zero, distinguishes from a full-page image */
uint32_t checksum;
```

Reply:
My initial plan is that we will still use the WT_BLOCK_HEADER for delta. I'll leave the checksum aside for now. If there is change of plan, we can modify this.

Reply:
@ddanderson In the current implementation, I leave the space for the block header. If we don't want the block header for delta, we can change here.
```
/*
 * __rec_build_delta_init --
 *     Build delta init.
 */
static int
__rec_build_delta_init(WT_SESSION_IMPL *session, WT_RECONCILE *r)
{
    WT_RET(__wt_buf_init(session, &r->delta, r->disk_img_buf_size));
    memset(r->delta.mem, 0, WT_DELTA_HEADER_SIZE);
    r->delta.size = WT_DELTA_HEADER_BYTE_SIZE(S2BT(session));

    return (0);
}
```